packages/client-sdk-nodejs/src/batchutils/batch-functions.ts

```diff
@@ -0,0 +1,145 @@
+import {CacheDelete, CacheGet, CacheSet} from '@gomomento/sdk-core';
+import {CacheClient} from '../cache-client';
+import {
+  BatchDeleteRequest,
+  BatchDeleteResponse,
+  BatchGetRequest,
+  BatchGetResponse,
+  BatchSetItem,
+  BatchSetRequest,
+  BatchSetResponse,
+  defaultMaxConcurrentRequests,
+} from './batch-props';
+import {range} from '@gomomento/sdk-core/dist/src/internal/utils';
+
+export {
+  BatchDeleteRequest,
+  BatchDeleteResponse,
+  BatchGetRequest,
+  BatchGetResponse,
+  BatchSetItem,
+  BatchSetRequest,
+  BatchSetResponse,
+  defaultMaxConcurrentRequests,
+} from './batch-props';
+
+export async function batchGet(
+  request: BatchGetRequest
+): Promise<BatchGetResponse> {
+  const maxConcurrentGets = request.maxConcurrentGets
+    ? request.maxConcurrentGets
+    : Math.min(defaultMaxConcurrentRequests, request.keys.length);
+
+  const batchGetResults = range(maxConcurrentGets).map((workerId: number) =>
+    getWorker(workerId, request.cacheClient, request.cacheName, request.keys)
+  );
+  const awaitAll = await Promise.all(batchGetResults);
+
+  const batchGetResponse: BatchGetResponse = {};
+  awaitAll.forEach(responses => {
+    for (const key of Object.keys(responses)) {
+      batchGetResponse[key] = responses[key];
+    }
+  });
+  return batchGetResponse;
+}
+
+async function getWorker(
+  workerId: number,
+  cacheClient: CacheClient,
+  cacheName: string,
+  keys: Array<string>
+): Promise<Record<string, CacheGet.Response>> {
+  const responses: Record<string, CacheGet.Response> = {};
+  while (keys.length) {
+    const cacheKey = keys.pop();
+    if (cacheKey) {
```

Comment:
nit/thought about JS semantics - JS has a really flexible definition of `truthiness`. Lots of things like `0` and a boolean `false` would fail this condition, so it's probably best to read the docs for `pop` and explicitly test for what it returns when the list is empty, which would guess is going to either be `null` or `undefined`.